roles/pulp_workers/tasks/main.yml

```
@@ -9,7 +9,36 @@
   notify: Reload systemd and restart pulp workers
   become: true
 
-- name: Set states of pulp workers
+- name: Get services
+  service_facts:
+
+- name: Get already existing workers
+  set_fact:
+    worker_services: '{{ worker_services|default([]) + [ item | regex_search("(?<=pulpcore-worker@)\d+(?=\.service$)") ] }}'
+  loop: "{{ ansible_facts.services.keys() | list }}"
+  when:
+    - item | regex_search("pulpcore-worker@\d+\.service")
```

```suggestion:
      loop: '{{ ansible_facts.services.keys() | list | select("match", "pulpcore-worker@\d+\.service") | list }}'
```
This will make the ansible output much shorter, it will only list the pulpcore services.

It may also speed up execution time heavily.